### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by erfanio**
Update packages
and fix tests

### CODE REVIEW

**Comment by erfanio** [comment]
Hmmm looks like travis config needs changing too

**Comment by erfanio** [comment]
@arb how does this look?

File: package.json [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
+    "moment": "2.22.x"
   },
   "devDependencies": {
-    "code": "4.x.x",
-    "lab": "14.x.x"
+    "code": "5.x.x",
+    "lab": "15.x.x"
   },
   "engines": {
     "node": ">=4.x.x"
```
Comment (arb) [inline package.json]: Make sure you update this too

**Comment by erfanio** [comment]
@arb done :rocket:

**Comment by arb** [comment]
So since this uses joi@13, do we need to copy it's node engine setting? https://github.com/hapijs/joi/blob/master/package.json#L13-L15

**Comment by erfanio** [comment]
Yeh we should

I just copied [Good's engine version](https://github.com/hapijs/good/blob/master/package.json#L16) which seems to need updating too.

It's really weird tho, npm doesn't give any warnings but yarn completely fails on node 8, seems like [npm is ignoring engine](https://github.com/npm/npm/issues/12486)

**Pull request closed without merging**